### ISSUE TITLE
Error: listen EADDRINUSE :::8989

### ISSUE THREAD

**Issue by lpar** [ssbc/patchwork#972]
Starting up Patchwork from command line AppImage, output:

```
[Usual stuff omitted]
ssb-friends: createFriendStream legacy api used
git-ssb is not installed, or not available in path
Error: listen EADDRINUSE :::8989
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at Server.setupListenHandle [as _listen2] (net.js:1351:14)
    at listenInCluster (net.js:1392:12)
    at Server.listen (net.js:1476:7)
    at EventEmitter.emitter.listen (/tmp/.mount_Patchwkulwvx/app/resources/app.asar/node_modules/pull-ws/server.js:50:14)
    at Object.server (/tmp/.mount_Patchwkulwvx/app/resources/app.asar/node_modules/scuttlebot/node_modules/multiserver/plugins/ws.js:20:31)
    at Object.server (/tmp/.mount_Patchwkulwvx/app/resources/app.asar/node_modules/multiserver/compose.js:96:29)
    at /tmp/.mount_Patchwkulwvx/app/resources/app.asar/node_modules/multiserver/index.js:45:21
    at Array.map (<anonymous>)
```

Fedora 29, Patchwork 3.11.6 downloaded just now.

**Comment by lpar** [comment]
3.11.5: Same error
3.11.4: No error, waiting to see if it works and content eventually appears

**Comment by lpar** [comment]
3.11.4 still works

**Comment by christianbundy** [comment]
Could you post the config that's being output on start? No need to omit, it's very useful for debugging.

**Comment by lpar** [comment]
```
{ path: '/home/meta/.ssb',
  party: true,
  timeout: 0,
  pub: true,
  local: true,
  friends: { dunbar: 150, hops: 2 },
  gossip: { connections: 3 },
  connections: 
   { outgoing: { net: [Array] },
     incoming: { net: [Array], ws: [Array] } },
  timers: { connection: 0, reconnect: 5000, ping: 300000, handshake: 5000 },
  caps: 
   { shs: '1KHLiKZvAvjbY1ziZEHMXawbCEIM6qwjCDm3VYRan/s=',
     sign: null },
  master: [],
  logging: { level: 'notice' },
  port: 8008,
  blobsPort: 8989,
  server: true,
  _: [],
  host: '9.85.238.89',
  ws: 
   { host: '127.0.0.1',
     port: 8989,
     scope: [ 'device' ],
     transform: 'shs' },
  keys: 
   { curve: 'ed25519',
     public: '5D5wZZjJGpUZ+DcgkoJXvseIeJKUkYK7G5JwcocBXA0=.ed25519',
     private: null,
     id: '@5D5wZZjJGpUZ+DcgkoJXvseIeJKUkYK7G5JwcocBXA0=.ed25519' } }
ssb-friends: createFriendStream legacy api used
git-ssb is not installed, or not available in path
```

**Comment by christianbundy** [comment]
Hmm... can you verify that you don't have a `remote` property? That's surprising to me, maybe that's where the bug is coming from.

**Comment by lpar** [comment]
Grepped the output for `remote`, no matches.

**Comment by PrimordialHelios** [comment]
I get the same error on Windows 10 with 3.11.6
```
{ path: 'C:\\Users\\tay\\.ssb',
  party: true,
  timeout: 0,
  pub: true,
  local: true,
  friends: { dunbar: 150, hops: 2 },
  gossip: { connections: 3 },
  connections:
   { outgoing: { net: [Array] },
     incoming: { net: [Array], ws: [Array] } },
  timers: { connection: 0, reconnect: 5000, ping: 300000, handshake: 5000 },
  caps:
   { shs: '1KHLiKZvAvjbY1ziZEHMXawbCEIM6qwjCDm3VYRan/s=',
     sign: null },
  master: [],
  logging: { level: 'notice' },
  port: 8008,
  blobsPort: 8989,
  server: true,
  _: [],
  host: '::',
  ws: { host: '::1', port: 8989, scope: [ 'device' ], transform: 'shs' },
  keys:
   { curve: 'ed25519',
     public: 'KSUfZY6SPENeJ98s5bunjdJyjCWBIzgGiUNdGHowP80=.ed25519',
     private: null,
     id: '@KSUfZY6SPENeJ98s5bunjdJyjCWBIzgGiUNdGHowP80=.ed25519' } }
ssb-friends: createFriendStream legacy api used
git-ssb is not installed, or not available in path
Error: listen EADDRINUSE :::8989
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at Server.setupListenHandle [as _listen2] (net.js:1351:14)
    at listenInCluster (net.js:1392:12)
    at Server.listen (net.js:1476:7)
    at EventEmitter.emitter.listen (C:\Users\tay\AppData\Local\Programs\ssb-patchwork\resources\app.asar\node_modules\pull-ws\server.js:50:14)
    at Object.server (C:\Users\tay\AppData\Local\Programs\ssb-patchwork\resources\app.asar\node_modules\scuttlebot\node_modules\multiserver\plugins\ws.js:20:31)
    at Object.server (C:\Users\tay\AppData\Local\Programs\ssb-patchwork\resources\app.asar\node_modules\multiserver\compose.js:96:29)
    at C:\Users\tay\AppData\Local\Programs\ssb-patchwork\resources\app.asar\node_modules\multiserver\index.js:45:21
    at Array.map (<anonymous>)
Error: listen EADDRINUSE :::8008
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at Server.setupListenHandle [as _listen2] (net.js:1351:14)
    at listenInCluster (net.js:1392:12)
    at doListen (net.js:1501:7)
    at _combinedTickCallback (internal/process/next_tick.js:141:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
Error: no source:patchwork,channels
    at Object.localCall (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\muxrpc\local-api.js:29:13)
    at Object.<anonymous> (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\muxrpc\local-api.js:37:22)
    at PacketStreamSubstream.stream.read (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\muxrpc\stream.js:69:23)
    at PacketStream._onstream (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\packet-stream\index.js:228:11)
    at PacketStream.write (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\packet-stream\index.js:135:41)
    at C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\muxrpc\pull-weird.js:56:15
    at C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\pull-stream\sinks\drain.js:24:37
    at C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\pull-goodbye\node_modules\pull-stream\throughs\filter.js:17:11
    at Object.cb (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\packet-stream-codec\index.js:111:11)
    at drain (C:\Users\tay\AppData\Roaming\npm\node_modules\ssb-server\node_modules\pull-reader\index.js:39:14)
```

**Comment by PrimordialHelios** [comment]
Not sure if this is at all related, but I get the same error on Windows 10 if I start `ssb-server` manually before opening patchwork.

**Comment by Wieke** [comment]
I seem to be having this issue as well. I've noticed about it:

* Removing ~/.ssb doesn't help. You still see the EADDRINUSE error still happens, so it seems unlikely that config is the issue. Unless there's config stored in other places.
* Using `sudo fuser 8989/tcp` and `sudo fuser 8989/udp` I checked if the port was perhaps already in use before starting patchwork, they weren't. Once patchwork runs/hangs the ports are in use by `ssb-patchwork`, which makes sense.

**Comment by christianbundy** [comment]
I've pushed a new branch to Patchwork called `use-ipv4` that uses the code from [this PR](https://github.com/ssbc/ssb-config/pull/47), any chance it resolves this issue?

**Comment by lpar** [comment]
It made sense that it might be IPv6-related, since my network is IPv6 first, but it seems not. I just built `origin/HEAD` and that doesn't have this specific problem, though I get another unrelated error:

```
Error: expected buffer, found string
    at /home/meta/WIP/patchwork/node_modules/flumeview-hashtable/index.js:39:43
    at /home/meta/WIP/patchwork/node_modules/atomic-file/inject.js:46:9
    at FSReqWrap.readFileAfterClose [as oncomplete] (fs.js:511:3)
```

It then seems to sit there part-way through "Indexing database" forever; I've given it 15 minutes. It detects updates, but since it never finishes reindexing, I can't actually read any of them.

`use-ipv4` behaves identically, including that error.

**Comment by christianbundy** [comment]
@lpar I think that's an unrelated error (https://github.com/flumedb/flumeview-hashtable/issues/2), could you try restarting Patchwork and checking to see whether it fixes the error?

Also, I want to make sure that I understand this bit:

> I just built origin/HEAD and that doesn't have this specific problem

Does this mean that the EADDRINUSE problem is resolved in `master`?

**Comment by lpar** [comment]
Yes, the EADDRINUSE problem is resolved in master as far as I can tell. (Unless it's a race condition perhaps.)

**Comment by matthiasbeyer** [comment]
Any new release with this upcoming in the next days? Like 3.14.2 for example?

**Comment by christianbundy** [comment]
The fix was been released in 3.13.0, if you're still having this problem in 3.14.1 please create a new issue. Thank you!
